scripts/fix-lint.sh

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 MongoDB Inc
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+STAGED_GO_FILES=$(git diff --name-only | grep ".go$" | grep -v "mock")
+
+echo "==> Fixing lint errors..."
+for FILE in ${STAGED_GO_FILES}; do
+    golangci-lint run --fix --timeout 5m "${FILE}"
```

Comment:
do you know what config this uses, I'm worry this is not seeing the lint config since it's a different folder, can you confirm this is not using some default config instead of our file? adding a `--debug` here may help

Reply:
it is using the correct config `INFO [config_reader] Used config file .golangci.yml`

```
make lint
==> Linting changed go files...
INFO [config_reader] Config search paths: [./ /Users/andrea.angiolillo/workspace/mongocli /Users/andrea.angiolillo/workspace /Users/andrea.angiolillo /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 40 linters: [bodyclose deadcode depguard dogsled errcheck exhaustive exportloopref gochecknoinits goconst gocritic gocyclo godot gofmt goimports gomnd goprintffuncname gosec gosimple govet ifshort ineffassign makezero misspell nakedret noctx nolintlint predeclared revive rowserrcheck staticcheck structcheck stylecheck tenv thelper typecheck unconvert unparam unused varcheck whitespace]
INFO [loader] Using build tags: [e2e unit integration]
INFO [loader] Go packages loading at mode 575 (imports|name|compiled_files|exports_file|files|deps|types_sizes) took 1.119641371s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 303.301µs
INFO [linters context/goanalysis] analyzers took 775.471008ms with top 10 stages: gocritic: 586.460513ms, buildir: 45.910149ms, buildssa: 40.132328ms, goimports: 24.443687ms, gofmt: 23.454552ms, the_only_name: 22.613306ms, unparam: 3.319771ms, predeclared: 2.060592ms, nilness: 1.903206ms, fact_deprecated: 1.870302ms
INFO [runner] Issues before processing: 8, after processing: 1
INFO [runner] Processors filtering stat (out/in): identifier_marker: 8/8, exclude-rules: 2/8, nolint: 2/2, uniq_by_line: 1/2, source_code: 1/1, severity-rules: 1/1, path_prettifier: 8/8, max_per_file_from_linter: 1/1, max_same_issues: 1/1, max_from_linter: 1/1, path_shortener: 1/1, exclude: 8/8, skip_dirs: 8/8, autogenerated_exclude: 8/8, diff: 1/1, path_prefixer: 1/1, skip_files: 8/8, filename_unadjuster: 8/8, sort_results: 1/1, cgo: 8/8
INFO [runner] processing took 672.556µs with stages: exclude-rules: 206.337µs, nolint: 176.798µs, identifier_marker: 128.324µs, autogenerated_exclude: 61.181µs, source_code: 46.084µs, path_prettifier: 34.542µs, skip_dirs: 8.061µs, max_per_file_from_linter: 3.173µs, cgo: 2.184µs, uniq_by_line: 1.577µs, max_same_issues: 1.027µs, filename_unadjuster: 648ns, path_shortener: 627ns, max_from_linter: 624ns, exclude: 451ns, diff: 235ns, skip_files: 229ns, severity-rules: 185ns, sort_results: 152ns, path_prefixer: 117ns
INFO [runner] linters took 805.429219ms with stages: goanalysis_metalinter: 804.688824ms
internal/cli/atlas/backup/list.go:32: File is not `gofmt`-ed with `-s` (gofmt)
				cli.ListOpts
INFO File cache stats: 2 entries of total size 5.7KiB
INFO Memory: 24 samples, avg is 75.5MB, max is 160.2MB
INFO Execution took 2.238832085s
==> Done...
```

Reply:
```
 make fix-lint
==> Fixing lint errors...
INFO [config_reader] Config search paths: [./ /Users/andrea.angiolillo/workspace/mongocli /Users/andrea.angiolillo/workspace /Users/andrea.angiolillo /Users /]
INFO [config_reader] Used config file .golangci.yml
INFO [lintersdb] Active 40 linters: [bodyclose deadcode depguard dogsled errcheck exhaustive exportloopref gochecknoinits goconst gocritic gocyclo godot gofmt goimports gomnd goprintffuncname gosec gosimple govet ifshort ineffassign makezero misspell nakedret noctx nolintlint predeclared revive rowserrcheck staticcheck structcheck stylecheck tenv thelper typecheck unconvert unparam unused varcheck whitespace]
INFO [loader] Using build tags: [e2e unit integration]
INFO [loader] Go packages loading at mode 575 (compiled_files|files|name|deps|exports_file|imports|types_sizes) took 1.094793748s
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 313.314µs
INFO [linters context/goanalysis] analyzers took 0s with no stages
INFO [runner] Issues before processing: 8, after processing: 2
INFO [runner] Processors filtering stat (out/in): diff: 2/2, skip_files: 8/8, identifier_marker: 8/8, uniq_by_line: 2/2, max_from_linter: 2/2, path_prefixer: 2/2, autogenerated_exclude: 8/8, nolint: 2/2, max_same_issues: 2/2, source_code: 2/2, filename_unadjuster: 8/8, path_prettifier: 8/8, skip_dirs: 8/8, exclude: 8/8, exclude-rules: 2/8, max_per_file_from_linter: 2/2, path_shortener: 2/2, severity-rules: 2/2, cgo: 8/8, sort_results: 2/2
INFO [runner] processing took 706.876µs with stages: exclude-rules: 192.778µs, nolint: 183.668µs, identifier_marker: 136.28µs, path_prettifier: 63.627µs, autogenerated_exclude: 60.235µs, source_code: 51.034µs, skip_dirs: 8.695µs, cgo: 4.384µs, path_shortener: 2.298µs, filename_unadjuster: 685ns, uniq_by_line: 651ns, max_same_issues: 622ns, max_per_file_from_linter: 404ns, max_from_linter: 379ns, skip_files: 213ns, sort_results: 212ns, exclude: 198ns, diff: 192ns, severity-rules: 190ns, path_prefixer: 131ns
INFO [runner] linters took 139.121808ms with stages: goanalysis_metalinter: 138.3425ms
INFO Line 32 has multiple issues but at least one of them isn't inline: []result.Issue{result.Issue{FromLinter:"gofmt", Text:"File is not `gofmt`-ed with `-s`", Severity:"", SourceLines:[]string{"\t\t\t\tcli.ListOpts"}, Replacement:(*result.Replacement)(0xc0004f0600), Pkg:(*packages.Package)(0xc000b51040), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"internal/cli/atlas/backup/list.go", Offset:0, Line:32, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""}, result.Issue{FromLinter:"goimports", Text:"File is not `goimports`-ed", Severity:"", SourceLines:[]string{"\t\t\t\tcli.ListOpts"}, Replacement:(*result.Replacement)(0xc0004f0630), Pkg:(*packages.Package)(0xc000b51040), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"internal/cli/atlas/backup/list.go", Offset:0, Line:32, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""}}
INFO Fix issue &result.Issue{FromLinter:"gofmt", Text:"File is not `gofmt`-ed with `-s`", Severity:"", SourceLines:[]string{"\t\t\t\tcli.ListOpts"}, Replacement:(*result.Replacement)(0xc0004f0600), Pkg:(*packages.Package)(0xc000b51040), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"internal/cli/atlas/backup/list.go", Offset:0, Line:32, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""} with range {32 32}
INFO fixer took 1.442634ms with stages: all: 1.442634ms
INFO File cache stats: 1 entries of total size 2.8KiB
INFO Memory: 17 samples, avg is 51.4MB, max is 66.0MB
INFO Execution took 1.542213648s
==> Done...
```